### DISCUSSION TITLE
How to add Comments

### DISCUSSION THREAD

**Discussion by MichaelGaoX** [OrchardCMS/OrchardCore#15880] [Q&A]
In orchard,it contains Managing Comments etc ,but I coulden't find comments module in orchardcore
http://docs.orchardproject.net/en/latest/Documentation/Moderating-comments/

**Comment by pgy866** [comment]
https://github.com/agriffard/Disqus.OrchardCore

https://www.nuget.org/packages/Disqus.OrchardCore

**Comment by MichaelGaoX** [comment]
I only use Beta 3 release,the nuget，and add the disqus to my contentType, create the new  contentType,it show the 'Show comments' ,then i checked,but when i published,I didn't find somewhere to add comment

**Comment by agriffard** [comment]
As explained in the Readme, did you create the site on Disqus and set the corresponding setting in the admin?

> Edit the settings of the Disqus Part and enter the name of the site that you created at https://disqus.com/.

**Comment by sebastienros** [comment]
@MatthijsKrempel suggested he would start working on it. If he hasn't progressed on it yet then feel free to start the module. It should be ported from O1 as all its features and its architecture should be the same.

**Comment by MatthijsKrempel** [comment]
@sebastienros swamped here with other duties :(

**Comment by MichaelGaoX** [comment]
@pgy866 @agriffard @sebastienros  I make it, just forget to say thank to you.In recent days,I am busy in bulid the po file about translating Orchard to Chinese.Thanks again

**Comment by pgy866** [comment]
@MichaelGaoX

Regarding localization, it is still too early,  can wait for RC, hope to contribute to this https://crowdin.com/project/orchard-core
`关于本地化，目前尚可太早，可以等待RC之后，希望贡献到此https://crowdin.com/project/orchard-core`

**Comment by sebastienros** [comment]
We could also think about performance by splitting the comments into separate documents. The main contentitem would always have the last N comments, but the rest would be handled by independent documents. Or extenalize the comments completely from the content item.

Also comments are threaded, does a comment contain all its answers? (or just the last N ones)

**Comment by RockNHawk** [comment]
Is there an self hosted comments module ? @sebastienros

**Comment by sebastienros** [comment]
@RockNHawk still not. Feel free to work on it if you are interested. I laid down some rough design in the previous comment.

**Comment by RockNHawk** [comment]
Okay, thank you, Orchard 1.x version has a Comments module, maybe I or someone can re-target it to Orchard Core, haha.
